src/main/java/org/jboss/modules/AliasModuleSpec.java

```diff
@@ -22,21 +22,58 @@
 
 package org.jboss.modules;
 
+import java.lang.instrument.ClassFileTransformer;
+import java.security.PermissionCollection;
+import java.util.Collections;
+import java.util.Map;
+
+import org.jboss.modules.filter.PathFilters;
+
 /**
  * A module specification for a module alias.
+ * <p>
+ * Note that because of MODULES-241, alias modules are handled as a regular modules with a single dependency on the alias
+ * target. Use {@link #asConcreteModuleSpec()} to get the regular {@link ConcreteModuleSpec} representing this
+ * {@link AliasModuleSpec}.
  *
  * @author <a href="mailto:david.lloyd@redhat.com">David M. Lloyd</a>
+ * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
  */
 public final class AliasModuleSpec extends ModuleSpec {
 
     private final ModuleIdentifier aliasTarget;
+    private final ConcreteModuleSpec concreteModuleSpec;
 
     AliasModuleSpec(final ModuleIdentifier moduleIdentifier, final ModuleIdentifier aliasTarget) {
         super(moduleIdentifier);
         this.aliasTarget = aliasTarget;
+
+        DependencySpec aliasTargetDependency = DependencySpec.createModuleDependencySpec(PathFilters.acceptAll(),
+                PathFilters.acceptAll(), null, aliasTarget, false);
+
+        final String mainClass = null;
+        final AssertionSetting assertionSetting = AssertionSetting.INHERIT;
+        final ResourceLoaderSpec[] resourceLoaders = new ResourceLoaderSpec[0];
```

Comment:
Moving package-private constants is fine.  I think ResourceLoaderSpec is a reasonable place for this.